src/utils/routing.js

```diff
@@ -0,0 +1,48 @@
+import page from 'page';
+
+import { store, buildStoreURLValue } from '../state/store';
+
+function updateStoreURL(absoluteURL) {
+  const storeURLValue = buildStoreURLValue(absoluteURL);
+  store.setField('url', storeURLValue);
+}
+
+export function makeAbsolute(path) {
+  const currentURL = new URL(window.location.href);
+  if (!path.startsWith(currentURL.origin)) {
+    return `${currentURL.origin}${path}${currentURL.search}${currentURL.hash}`;
+  }
+  return `${path}${currentURL.search}${currentURL.hash}`;
+}
+
+// Inspired by https://stackoverflow.com/a/52809105/4297741
+export function registerStoreURLUpdaters() {
+  // Navigation
+  window.history.pushState = ((f) => function pushState(...args) {
+    const ret = f.apply(this, args);
+    const state = args[0];
+    updateStoreURL(makeAbsolute(state.path));
+    window.dispatchEvent(new Event('pushstate'));
+    return ret;
+  })(window.history.pushState);
+
+  // Back/Forward buttons
+  window.addEventListener('popstate', (evt) => {
+    if (evt && evt.state) {
+      updateStoreURL(makeAbsolute(evt.state.path));
+    }
+  });
+}
+
+export function probeURL({ product = 'firefox', name, view = 'explore' } = {}, preserveQuery = true, preserveHash = true) {
+  const currentURL = new URL(window.location.href);
+  let nextURL = `/probe/${product}/${name}/${view}`;
```

Comment:
Depending on the remaining work appetite for Data Tools on this, this suggestion would be the one thing we should agree upon no matter what since it's technically the only bit of UX this PR introduces.